.azure-pipelines-templates/publish_build.yml

```diff
@@ -1,7 +1,4 @@
 
-parameters:
```

Comment:
Minor: Why have we lost these `parameters`? They're a useful way to document the expected names, types, defaults for parameters, though it seems we can remove them with no consequence so maybe this is fine?

Reply:
They are not lost, they are passed into the job

Reply:
I know they're still passed, but this declaration gives the expected signature. If we list all the expected parameters here, we should get some clearer errors next time we change `matrix.yml` (foo isn't passed, foo isn't a string, etc).